index.js

```diff
@@ -35,7 +35,7 @@ function fastifyView (fastify, opts, next) {
   const defaultCtx = opts.defaultContext || {}
   const layoutFileName = opts.layout
 
-  if (layoutFileName && type !== 'handlebars' && type !== 'ejs') {
+  if (layoutFileName && type !== 'handlebars' && type !== 'ejs' && type !== 'eta') {
```

Comment:
Seems like the error message is not revelant anymore.

Reply:
I think it's there because only Handlebars, EJS, and Eta support the `layout` option (support is added w/ the `withLayout` function).

Reply:
Could you just edit the error message after this condition to be more relevant?

Reply:
@zekth just updated the error message, let me know what you think :)